agent/engine/docker_client.go

```diff
@@ -158,6 +161,14 @@ type DockerClient interface {
 	// RemoveVolume removes a volume by its name. A timeout value should be provided for the request
 	RemoveVolume(string, time.Duration) error
 
+	// ListPluginsWithFilters returns the set of docker plugins installed on the host, filtered by options provided.
+	// A timeout value should be provided for the request.
+	ListPluginsWithFilters(bool, []string, time.Duration) ([]string, error)
```

Comment:
Just for my own understanding, what are the use-cases for invoking `ListPluginsWithFilters(false, ...)`?

Reply:
this is where capability=volumedriver (and enabled=true) filter is supposed to be applied, but go-dockerclient's ListPlugins currently doesn't handle filters. I am going to submit PR for this fix but didn't want to block our work.

Reply:
ah, makes sense. so, more for the sake of completion than anything else.